monai/data/utils.py

```diff
@@ -945,6 +948,55 @@ def __init__(self, even_divisible: bool = True, *args, **kwargs):
             self.total_size = data_len
 
 
+class DistributedWeightedRandomSampler(DistributedSampler):
```

Comment:
This file is now getting rather large, we should look later into splitting out the definitions here into other categories if that makes sense but for now maybe the samples should go into `samplers.py` in this directory.

Reply:
OK, sounds good.